language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/AddToConfigTomlCodeAction.java

```diff
@@ -0,0 +1,291 @@
+            return Collections.emptyList();
+        }
+
+        ConfigurableFinder.ModuleConfigDetails moduleConfigDetails = ConfigurableFinder.getModuleConfigDetails(context);
+        Map<String, ConfigurableFinder.ConfigVariable> configVarMap = moduleConfigDetails.configVariables();
+        VariableSymbol variableSymbol = (VariableSymbol) confSymbol.get();
+        ConfigurableFinder.ConfigVariable selectedConf = configVarMap.get(variableSymbol.getName().get());
+
+        Path configTomlPath = context.workspace().projectRoot(context.filePath()).resolve(CONFIG_TOML);
+        if (!configTomlPath.toFile().exists()) {
+            if (configVarMap.isEmpty()) {
+                return Collections.emptyList();
+            }
+            return withoutConfigTomlCodeActions(basicType, anyDataOrJsonType, moduleConfigDetails,
+                    selectedConf, configTomlPath);
+        }
+
+        Optional<Toml> toml = readToml(configTomlPath);
+        if (toml.isEmpty()) {
+            return Collections.emptyList();
+        }
+        Module module = context.currentModule().get();
+        int lastNodeLine = module.isDefaultModule() ? rootModuleConfigDiff(toml.get(),
+                orgName, module.moduleName().toString(), configVarMap)
+                : nonRootModuleConfigDiff(toml.get(), orgName, module.moduleName(), configVarMap);
+
+        if (configVarMap.isEmpty() || !configVarMap.containsKey(selectedConf.name())) {
+            return Collections.emptyList();
+        }
+
+        return withConfigTomlCodeActions(basicType, anyDataOrJsonType, moduleConfigDetails, selectedConf,
+                configTomlPath, toml.get(), module, lastNodeLine);
+    }
+
+    private List<CodeAction> withConfigTomlCodeActions(TypeSymbol basicType, TypeSymbol anyDataOrJsonType,
+                                                       ConfigurableFinder.ModuleConfigDetails moduleConfigDetails,
+                                                       ConfigurableFinder.ConfigVariable selectedConf,
+                                                       Path configTomlPath, Toml toml,
+                                                       Module module, int lastNodeLine) {
+        Position position;
+        boolean hasAtLeastOneEntry = lastNodeLine != 0;
+        if (lastNodeLine == 0 && !module.isDefaultModule()) {
+            Map<String, TopLevelNode> entries = toml.rootNode().entries();
+            for (TopLevelNode node : entries.values()) {
+                int nodeLine = node.location().lineRange().endLine().line();
+                if (nodeLine > lastNodeLine) {
+                    lastNodeLine = nodeLine;
+                }
+            }
+            position = PositionUtil.toPosition(
+                    LinePosition.from(lastNodeLine == 0 ? 0 : lastNodeLine + 3, 0));
+        } else {
+            position = PositionUtil.toPosition(
+                    LinePosition.from(lastNodeLine == 0 ? 0 : lastNodeLine + 1, 0));
+        }
```

Comment:
Since we do here a position calculation, How about we extract this logic into a separate function?